craft_parts/plugins/maven_plugin.py

```diff
@@ -205,5 +206,5 @@ def _create_settings(settings_path: Path) -> None:
 
 
 def _get_no_proxy_string() -> str:
-    no_proxy = [k.strip() for k in os.environ.get("no_proxy", "localhost").split(",")]
+    no_proxy = [k.strip() for k in os.environ.get("NO_PROXY", "localhost").split(",")]
```

Comment:
how come?

Reply:
Env usually request to be uppercase. And you get an error for that.

Reply:
Please double check if the name is not lowercase for a reason, and in either case we should also keep checking the lowercase version for backwards compatibility.

Reply:
this should be lowercase, you will break launchpad if you change this, it is a client interface